src/schema.ts

```diff
@@ -51,7 +54,29 @@ export class Schema {
       };
     });
 
-    return new Schema(fieldSchemas);
+    let schema = new Schema(fieldSchemas);
+
+    // calculate preset bins
+    for (let fieldSchema of fieldSchemas) {
+      if (fieldSchema.type === Type.QUANTITATIVE) {
+        fieldSchema.binStats = {};
+        for (let maxbins of opt.maxBinsList) {
+          fieldSchema.binStats[maxbins] = binSummary(maxbins, fieldSchema.stats);
+        }
+      } else if (fieldSchema.type === Type.TEMPORAL) {
+        // need to get min/max of date data
+        const millis = [];
+        for (var i = 0; i < data.length; i++) {
+          millis.push(new Date(data[i][fieldSchema.field]).getTime());
+        }
+        millis.sort();
```

Comment:
Sorting takes O(n log n). You can get min max in O(n).   (Just check if value is less than min and more than max in the loop.